packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -556,7 +576,20 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             }
         }
 
-        const registry = new ContainerRuntimeDataStoreRegistry(registryEntries);
+        const backCompatRuntimeOptions = getBackCompatRuntimeOptions(runtimeOptions);
+        const defaultRuntimeOptions: Required<IContainerRuntimeOptions> = {
+            summaryOptions: { generateSummaries: true },
+            gcOptions: {},
+            omitAgentSchedulerAndLeaderElection: false,
+        };
+        const combinedRuntimeOptions = { ...defaultRuntimeOptions, ...backCompatRuntimeOptions };
+        if (!combinedRuntimeOptions.omitAgentSchedulerAndLeaderElection) {
+            console.error("ContainerRuntime with AgentScheduler built-in is deprecated");
+        }
+
+        const registry = combinedRuntimeOptions.omitAgentSchedulerAndLeaderElection
```

Comment:
i think this will need to be treated more like isolated channels where it can only be set at container create, and is permanent. this isn't something that can be turned on and off as it affects the file contents

Reply:
Chatted with Tony -- we'll need to be careful here (similar document compatibility risk to changing the registry after creation in other scenarios) but not planning to persist the setting in the same way that isolated channels are.